sycl/source/detail/graph_impl.cpp

```diff
@@ -676,9 +676,12 @@ exec_graph_impl::enqueueNodeDirect(sycl::context Ctx,
   ur_result_t Res = sycl::detail::enqueueImpCommandBufferKernel(
       Ctx, DeviceImpl, CommandBuffer,
       *static_cast<sycl::detail::CGExecKernel *>((Node->MCommandGroup.get())),
-      Deps, &NewSyncPoint, &NewCommand, nullptr);
+      Deps, &NewSyncPoint, this->MIsUpdatable ? &NewCommand : nullptr, nullptr);
 
-  MCommandMap[Node] = NewCommand;
+  if(this->MIsUpdatable){
```

Comment:
nitpick, but can we drop the `this` from these `this->MIsUpdatable` expressions, not sure if it's convention or not.

Reply:
Removed, it seems like our usage is inconsistent but this is probably bad practice. I've left the other instances alone for now to not clutter this PR with unrelated changes.